internal/wasm/jit/asm/amd64/impl.go

```diff
@@ -0,0 +1,382 @@
+package amd64
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/tetratelabs/wazero/internal/wasm/jit/asm"
+)
+
+// nodeImpl implements asm.Node for amd64.
+type nodeImpl struct {
+	instruction asm.Instruction
+
+	// offsetInBinary represents the offset of this node in the final binary.
+	offsetInBinary int64
+	// jumpTarget holds the target node in the linked for the jump-kind instruction.
+	jumpTarget *nodeImpl
+	// prev holds the previous node to this node in the assembled linked list.
+	prev *nodeImpl
+	// next holds the next node from this node in the assembled linked list.
+	next *nodeImpl
+
+	types                    operandTypes
+	srcReg, dstReg           asm.Register
+	srcConst, dstConst       int64
```

Comment:
similar comment needed saying why 64 and why signed. actually this is everywhere so you can make a type alias to describe and use that to reduce comments. here and where they are used in parameters.

ex type sourcePosition = int64 (or something, then comment once)